src/server/req_rescq.c

```diff
@@ -351,9 +351,21 @@ remove_node_from_resv(resc_resv *presv, struct pbsnode *pnode)
 void
 remove_host_from_resv(resc_resv *presv, char *hostname) {
 	pbsnode_list_t *pl = NULL;
-	for (pl = presv->ri_pbsnode_list; pl != NULL; pl = pl->next) {
-		if (strcmp(pl->vnode->nd_hostname, hostname) == 0)
+	pbsnode_list_t *prev = NULL;
+
+	for (prev = NULL, pl = presv->ri_pbsnode_list; pl != NULL;) {
+		if (strcmp(pl->vnode->nd_hostname, hostname) == 0) {
 			remove_node_from_resv(presv, pl->vnode);
+			if (prev == NULL) {
```

Comment:
I don't understand this change. "prev" is never assigned a value so it will always be NULL. That means you are always removing from the front of the list.

Reply:
i had removed pl = pl->next from the for statement, to add it, but forgot. thanks for catching it.